libs/pavex_session_redis/src/lib.rs

```diff
@@ -0,0 +1,380 @@
+                )));
+            }
+        };
+
+        Ok(Some(SessionRecord { ttl, state }))
+    }
+
+    /// Deletes a session record from the store using the provided ID.
+    ///
+    /// If the session exists, it is removed from the store.
+    #[tracing::instrument(name = "Delete server-side session record", level = tracing::Level::INFO, skip_all)]
+    async fn delete(&self, id: &SessionId) -> Result<(), DeleteError> {
+        let ndeleted: u64 = self
+            .conn
+            .clone()
+            .del(self.redis_key(id))
+            .await
+            .map_err(|e| DeleteError::Other(e.into()))?;
+
+        if ndeleted == 1 {
+            Ok(())
+        } else if ndeleted == 0 {
+            Err(err_unknown(id).into())
+        } else {
+            return Err(DeleteError::Other(anyhow::anyhow!(
+                "Redis DEL replied {:?}. Expected 0 or 1.",
+                ndeleted
+            )));
+        }
+    }
+
+    /// Change the session id associated with an existing session record.
+    ///
+    /// The server-side state is left unchanged.
+    #[tracing::instrument(name = "Change id for server-side session record", level = tracing::Level::INFO, skip_all)]
+    async fn change_id(&self, old_id: &SessionId, new_id: &SessionId) -> Result<(), ChangeIdError> {
+        // Atomically check whether a key exists and then rename if it does.
+        const LUA_RENAME_IF_EXISTS: &str = r#"
+            if redis.call('EXISTS', KEYS[1]) == 1 then
+                -- returns 1 on success or 0 if fails because KEYS[2] already exists
+                return redis.call('RENAMENX', KEYS[1], KEYS[2])
+            else
+                return -1
+            end
+        "#;
+
+        let script = redis::Script::new(LUA_RENAME_IF_EXISTS);
+        let mut conn = self.conn.clone();
+        let old_key = self.redis_key(old_id);
+        let new_key = self.redis_key(new_id);
+        let result: i32 = script
+            .key(&old_key)
+            .key(&new_key)
+            .invoke_async(&mut conn)
+            .await
+            .map_err(|e| ChangeIdError::Other(e.into()))?;
+
+        match result {
+            -1 => Err(err_unknown(old_id).into()),  // Key didn't exist
+            1 => Ok(()),                            // Successfully renamed
+            0 => Err(err_duplicate(new_id).into()), // Key existed but new_key already exists (RENAMENX failed)
+            other => Err(ChangeIdError::Other(anyhow::anyhow!(
+                "Redis RENAMENX replied {:?}. Expected INT 0 or 1",
+                other
+            ))),
+        }
+    }
+
+    /// Deletes expired session records from the store.
+    ///
+    /// Redis handles the deletion of expired keys automatically, so this method always
+    /// returns Ok(0).
+    async fn delete_expired(
+        &self,
+        _batch_size: Option<NonZeroUsize>,
+    ) -> Result<usize, DeleteExpiredError> {
+        Ok(0)
+    }
+}
+
+pub struct RedisSessionKit {
```

Comment:
There's no longer a need for kits. It's enough to add a couple of attributes to get thing working with the new DI system.
You can use either pavex.dev/docs or the MySQL PR as a reference on this matter.